examples/seismic/viscoelastic/viscoelastic_example.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from argparse import ArgumentParser
+
+from devito.logger import info
+from examples.seismic.viscoelastic import ViscoelasticWaveSolver
+from examples.seismic import demo_model, AcquisitionGeometry
+
+
+def viscoelastic_setup(shape=(50, 50), spacing=(15.0, 15.0), tn=500.,
+                       space_order=4, nbpml=10,
+                       constant=True, **kwargs):
+
+    nrec = 2*shape[0]
+    preset = 'constant-viscoelastic' if constant else 'layers-viscoelastic'
+    model = demo_model(preset, space_order=space_order, shape=shape, nbpml=nbpml,
+                       dtype=kwargs.pop('dtype', np.float32), spacing=spacing)
+
+    # Source and receiver geometries
+    src_coordinates = np.empty((1, len(spacing)))
+    src_coordinates[0, :] = np.array(model.domain_size) * .5
+    if len(shape) > 1:
+        src_coordinates[0, -1] = model.origin[-1] + 2 * spacing[-1]
+
+    rec_coordinates = np.empty((nrec, len(spacing)))
+    rec_coordinates[:, 0] = np.linspace(0., model.domain_size[0], num=nrec)
+    if len(shape) > 1:
+        rec_coordinates[:, 1] = np.array(model.domain_size)[1] * .5
+        rec_coordinates[:, -1] = model.origin[-1] + 2 * spacing[-1]
+    geometry = AcquisitionGeometry(model, rec_coordinates, src_coordinates,
+                                   t0=0.0, tn=tn, src_type='Ricker', f0=0.12)
+
```

Comment:
We may want to make a `setup_gemoetry` in `utils.py` or something as this is the same setup code in all the seismic examples.

Reply:
Do we want to move this now or wait for a future update?

Reply:
can you do this in a separate PR?

I'd really like this